h/browser/chrome/lib/pdfHandler.js

```diff
@@ -0,0 +1,191 @@
+  var header = getHeaderFromHeaders(details.responseHeaders, 'content-type');
+  if (header) {
+    var headerValue = header.value.toLowerCase().split(';',1)[0].trim();
+    return (headerValue === 'application/pdf' ||
+            headerValue === 'application/octet-stream' &&
+            details.url.toLowerCase().indexOf('.pdf') > 0);
+  }
+}
+
+/**
+ * Takes a set of headers, and set "Content-Disposition: attachment".
+ * @param {Object} details First argument of the webRequest.onHeadersReceived
+ *                         event. The property "responseHeaders" is read and
+ *                         modified if needed.
+ * @return {Object|undefined} The return value for the onHeadersReceived event.
+ *                            Object with key "responseHeaders" if the headers
+ *                            have been modified, undefined otherwise.
+ */
+function getHeadersWithContentDispositionAttachment(details) {
+  var headers = details.responseHeaders;
+  var cdHeader = getHeaderFromHeaders(headers, 'content-disposition');
+  if (!cdHeader) {
+    cdHeader = {name: 'Content-Disposition'};
+    headers.push(cdHeader);
+  }
+  if (!/^attachment/i.test(cdHeader.value)) {
+    cdHeader.value = 'attachment' + cdHeader.value.replace(/^[^;]+/i, '');
+    return { responseHeaders: headers };
+  }
+}
+
+chrome.webRequest.onHeadersReceived.addListener(
+  function(details) {
+    // if we're not an active tab, do nothing
+    if (!state.isTabActive(details.tabId)) {
+      return;
+    }
+
+    if (details.method !== 'GET') {
+      // Don't intercept POST requests until http://crbug.com/104058 is fixed.
+      return;
+    }
+    if (!isPdfFile(details)) {
+      return;
+    }
+    if (isPdfDownloadable(details)) {
+      // Force download by ensuring that Content-Disposition: attachment is set
+      return getHeadersWithContentDispositionAttachment(details);
+    }
+
+    var viewerUrl = getViewerURL(details.url);
+
+    // Implemented in preserve-referer.js
+    saveReferer(details);
+
+    return { redirectUrl: viewerUrl };
+  },
+  {
+    urls: [
+      '<all_urls>'
+    ],
+    types: ['main_frame']
+  },
+  ['blocking','responseHeaders']);
+
+chrome.webRequest.onBeforeRequest.addListener(
+  function onBeforeRequestForFTP(details) {
+    // if we're not an active tab, do nothing
+    if (!state.isTabActive(details.tabId)) {
+      return;
+    }
+
+    if (isPdfDownloadable(details)) {
+      return;
+    }
+    var viewerUrl = getViewerURL(details.url);
+    return { redirectUrl: viewerUrl };
+  },
+  {
+    urls: [
+      'ftp://*/*.pdf',
+      'ftp://*/*.PDF'
+    ],
+    types: ['main_frame']
+  },
+  ['blocking']);
+
+chrome.webRequest.onBeforeRequest.addListener(
+  function(details) {
+    // if we're not an active tab, do nothing
+    if (!state.isTabActive(details.tabId)) {
+      return;
+    }
+
+    if (isPdfDownloadable(details)) {
+      return;
+    }
+
+    // NOTE: The manifest file has declared an empty content script
```

Comment:
The relevant bug is marked fixed since March 2014. This comment can probably go. It's not accurate for our packaging, anyway.